### DISCUSSION TITLE
Default RPC server vs HTTP server

### DISCUSSION THREAD

**Discussion by Davincible** [micro/go-micro#2577] [General] [answered]
@asim, the RPC server located in `server/rpc_*.go` and `plugins/server/http` are both HTTP servers, but the former seems to be much bigger than the latter.

Do you know if there is any big difference between the two, as they're both HTTP servers?

**Comment by asim** [comment, marked as answer]
Yea the http plugin is a hack to get standard http requests to work with the framework. The RPC implementation baked into go-micro is the original RPC code that's transport agnostic, has pluggable codecs, etc.

**Reply by Davincible** [reply]
Ahhh okay makes sense thanks
